lab-jacob/server.js

```diff
@@ -0,0 +1,53 @@
+'use stict';
+
+const cowsay = require('cowsay');
+const http = require('http');
+const urlParse = require('url').parse;
+const queryParse = require('querystring').parse;
+
+const bodyParse = require('./lib/bodyParse');
+const PORT = process.argv[2] || 3000;
+
+const server = http.createServer(function(req, res) {
+  req.url = urlParse(req.url); //parses the URL
+  req.url.query = queryParse(req.url.query); //parses the query from url
+
+  console.log('req.method', req.method);
+
+  if(/(PUT|POST|DELETE)/.test(req.method)){
```

Comment:
Making a POST request to '/cowsay' breaks the server, with the error of "write after end".  You are close, but please edit your server to have a response to endpoint /cowsay with POST.